### PR TITLE
Fix for incorrect table selection styles in readonly mode

### LINKED ISSUE #1887: Table Selection has incorrect styling in readonly editor
## Are you reporting a feature request or a bug?

Bug

## Provide detailed reproduction steps (if any)

1. Open https://jsfiddle.net/sjg91czp/
2. Select some cells.

### Expected result

Selection is styled like the editor has focus (blue background).

### Actual result

Selection is styled like the editor is blurred (gray background).

## Other details

* CKEditor version: 4.9.1
* Installed CKEditor plugins: tableselection

### PR DESCRIPTION
**Pull request by Comandeer**
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_contributing_code-section-tests),
[how to set the testing environment](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_tests) and
[how to create tests](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_tests-section-creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## What changes did you make?

I've added new clas, `.cke_table-faked-selection-focused-editor`, that is added to iframe-based editor when editor is focused. It is necessary as editor has issues with managing focus in readonly mode.

Closes #1887.

### CODE REVIEW

**Comment by Comandeer** [comment]
I've rebased branch to latest `master`. I've also fixed unit tests in Firefox as rebasing fixed tests in Edge.

Please also note that this PR fixes the issue only in classic editor as inline ones are already working – except IE 11 and Edge, probably due to https://github.com/ckeditor/ckeditor-dev/issues/1907

File: tests/plugins/tableselection/tableselection.js [hunk 1]
```diff
@@ -145,6 +145,30 @@
 			mockMouseSelection( editor, [ cells.getItem( 1 ), cells.getItem( 3 ), cells.getItem( 8 ) ], function() {
 				assert.pass();
 			} );
+		},
+
+		// (#1887)
+		'test toggling class while focusing/bluring editor': function( editor ) {
```
Comment (mlewand) [inline tests/plugins/tableselection/tableselection.js]: For this ticket I'd go with implementation-agnostic unit test - based on (computed) color of a cell's background.

**Comment by Comandeer** [comment]
Closing as incorrect solution; see https://github.com/ckeditor/ckeditor-dev/issues/1907#issuecomment-413512634
